From ced1149f5af28ce2ad6174a2df9483b483d5e1cf Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 23 Sep 2015 19:39:30 +0200
Subject: [PATCH] journal-cat: return a correct error, not -1

Cherry-picked from: e4603df5cf80bbd7a7d51fc66fa6c60e042423bc
Related: #1318994
---
 src/journal/cat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/journal/cat.c b/src/journal/cat.c
index 79706b692..ed467b078 100644
--- a/src/journal/cat.c
+++ b/src/journal/cat.c
@@ -96,7 +96,7 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_priority = log_level_from_string(optarg);
                         if (arg_priority < 0) {
                                 log_error("Failed to parse priority value.");
-                                return arg_priority;
+                                return -EINVAL;
                         }
                         break;
 
